### PR TITLE
feat(dashboards): Add management of favorite dashboards

### PR DESCRIPTION
**Pull request by Med1233**
## Description

The goal is to offer users a way to mark their favorite dashboards so they can easily retrieve them.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/cypress/visual-testing-snapshots/www/front_src/src/Dashboards/Dashboards.cypress.spec.tsx/Dashboards --  creator role Dashboard favorite -- View as cards toggles the dashboard favorite status when the favorite button is clicked.snap.png
Comment (Thebarda) [inline centreon/cypress/visual-testing-snapshots/www/front_src/src/Dashboards/Dashboards.cypress.spec.tsx/Dashboards --  creator role Dashboard favorite -- View as cards toggles the dashboard favorite status when the favorite button is clicked.snap.png:1]: As the favorite icon is always visible no matter the user ACL, can we agree to display it in the same place either at the bottom left, bottom right or top left of the card ? What do you think @Wedierre and @Med1233

Comment (Thebarda) [inline centreon/cypress/visual-testing-snapshots/www/front_src/src/Dashboards/Dashboards.cypress.spec.tsx/Dashboards --  creator role Dashboard favorite -- View as cards toggles the dashboard favorite status when the favorite button is clicked.snap.png:1]: As the favorite icon is always visible no matter the user ACL, can we agree to display it in the same place either at the bottom left, bottom right or top left of the card ? What do you think @Wedierre and @Med1233

File: centreon/cypress/visual-testing-snapshots/www/front_src/src/Dashboards/Dashboards.cypress.spec.tsx/Dashboards --  creator role Dashboard favorite -- View as list displays the favorite icons in the correct state.snap.png
Comment (Thebarda) [inline centreon/cypress/visual-testing-snapshots/www/front_src/src/Dashboards/Dashboards.cypress.spec.tsx/Dashboards --  creator role Dashboard favorite -- View as list displays the favorite icons in the correct state.snap.png:1]: Same comment here, the favorite should remain at the same position (the first position)

File: centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { useStyles } from './DataTableItem.styles';
 
 export interface DataTableItemProps {
   Actions?: JSX.Element;
+  fovoriteAction?: JSX.Element;
```
Comment (Thebarda) [inline centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx:15]:
```suggestion
```

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/Dashboard.cypress.spec.tsx [hunk 2]
```diff
@@ -11,57 +11,58 @@ import { BrowserRouter } from 'react-router-dom';
...
-    labelAddAContact,
-    labelDelete,
-    labelSharesSaved
+  labelAddAContact,
+  labelDelete,
+  labelSharesSaved
 } from '../../translatedLabels';
 
-import { dashboardAtom } from './atoms';
+import { profileEndpoint } from 'www/front_src/src/api/endpoint';
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/Dashboard.cypress.spec.tsx:38]: use relative import

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -94,6 +104,7 @@ const Dashboard = (): ReactElement => {
               description={dashboard?.description || ''}
               title={dashboard?.name || ''}
             />
+            { !isOnPublicPage && <Favorite dashboardId = {Number(dashboardId)} isFavorite = {isFavorite} />}
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/Dashboard.tsx:107]:
```suggestion
            { !isOnPublicPage && <Favorite dashboardId={Number(dashboardId)} isFavorite={isFavorite} />}
```

File: centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -82,6 +87,11 @@ const Dashboard = (): ReactElement => {
     };
   }, []);
 
+  const isFavorite = useMemo(
+    () => profile?.favoriteDashboards?.includes(Number(dashboardId)),
+    [profile]
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/Dashboard.tsx:92]:
```suggestion
    [profile, dashboardId]
```

File: centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx [hunk 5]
```diff
@@ -27,7 +28,8 @@ const DataTableItem = forwardRef(
       hasCardAction = false,
       hasActions = false,
       onClick,
-      Actions
+      Actions,
+      fovoriteAction
```
Comment (Thebarda) [inline centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx:32]:
```suggestion
```
Why did you create a prop for favorite actions as the Favorite icon is already in the Actions prop

File: centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx [hunk 6]
```diff
@@ -53,12 +55,10 @@ const DataTableItem = forwardRef(
             {description && <MuiTypography>{description}</MuiTypography>}
           </MuiCardContent>
         </ActionArea>
-        {hasActions && (
           <MuiCardActions>
             <span />
-            <span>{Actions}</span>
+            <span>{hasActions ? Actions : fovoriteAction}</span>
```
Comment (Thebarda) [inline centreon/packages/ui/src/components/DataTable/Item/DataTableItem.tsx:60]:
```suggestion
            <span>{Actions}</span>
```

File: centreon/www/front_src/src/Dashboards/components/DashboardLibrary/DashboardsOverview/DashboardsOverview.tsx [hunk 7]
```diff
@@ -82,7 +89,18 @@ const DashboardsOverview = (): ReactElement => {
...
+              dashboard={dashboard}
+              isFavorite={getIsFavorite(dashboard)}
+            />
+          }
+          fovoriteAction={
+            <Favorite
+              dashboardId={Number(dashboard?.id)}
+              isFavorite={getIsFavorite(dashboard)}
+            />
+          }
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/components/DashboardLibrary/DashboardsOverview/DashboardsOverview.tsx:103]:
```suggestion
```

**Review by Thebarda — APPROVED** [review]

**Review by quborg — APPROVED** [review]

**Review by quborg — APPROVED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -19468,8 +19468,17 @@ msgstr "Affiche l'heure en fonction du fuseau horaire sélectionné, ou un minut
...
-# msgid "Displays the time according to the selected time zone, or a timer."
-# msgstr ""
+msgid "Dashboard marked as favorite"
+msgstr "Tableau de bord marqué comme favori"
 
-# msgid "Clock / Timer"
-# msgstr ""
\ No newline at end of file
+msgid "Dashboard unmarked as favorite"
+msgstr "Tableau de bord désélectionné comme favori"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19475]:
```suggestion
msgstr "Tableau de bord non marqué comme favori"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -19468,8 +19468,17 @@ msgstr "Affiche l'heure en fonction du fuseau horaire sélectionné, ou un minut
...
+
+msgid "Mark as favorite"
+msgstr "Marquer comme favori"
+
+msgid "Unmark as favorite"
+msgstr "Désélectionner comme favori"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19481]:
```suggestion
msgstr "Ne plus marquer comme favori"
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Pull request merged**